### PR TITLE
PLANET-6968 Move some typography settings to theme.json

### PR DESCRIPTION
**Pull request by mleray**
### Description

See [PLANET-6968](https://jira.greenpeace.org/browse/PLANET-6968)
This is to simplify our code and follow WP guidelines. Options moved include:
- link colors
- heading font family and font weight
- body text/background color and font family

### Testing

You can make sure (on local or on the [sinope test instance](https://www-dev.greenpeace.org/test-sinope/)) that links and headings still look as expected. Note that you need WordPress 6.1 or up for the changes to work!

### CODE REVIEW

**Comment by planet-4** [comment]
### Test instance is ready :rocket:

:new_moon: [sinope](https://www-dev.greenpeace.org/test-sinope) | [admin](https://www-dev.greenpeace.org/test-sinope/wp-admin/) | [blocks report](https://www-dev.greenpeace.org/test-sinope/wp-admin/?page=plugin_blocks_report) | [CircleCI](https://app.circleci.com/pipelines/github/greenpeace/planet4-test-sinope) | [composer-local.json](https://github.com/greenpeace/planet4-test-sinope/blob/main/composer-local.json)

:watch: 2022.11.18 13:34:55

File: functions.php [hunk 1]
```diff
@@ -220,7 +220,7 @@ function ( $hidden ) {
 add_filter(
 	'block_editor_settings_all',
 	function ( array $editor_settings, WP_Block_Editor_Context $block_editor_context ) {
-		if ( 'post' !== $block_editor_context->post->post_type ) {
+		if ( isset( $block_editor_context->post ) && 'post' !== $block_editor_context->post->post_type ) {
```
Comment (mleray) [inline functions.php:223]: This is not really related to the ticket, but it's such a small change that I didn't think it required a separate PR

File: assets/src/scss/base/_typography.scss [hunk 2]
```diff
@@ -25,11 +25,6 @@ h3,
 h4,
 h5,
 h6 {
-  --headings-- {
-    font-family: $roboto;
-    font-weight: bold;
-  }
```
Comment (lithrel) [inline assets/src/scss/base/_typography.scss:31]: I see [some code using these variables](https://github.com/greenpeace/planet4-child-theme-indonesia/blob/8ac284d4adc714447502c8c119dd0b5bdbca486f/assets/src/scss/layout/_page-section.scss#L4-L5), I guess it won't have an impact because of default values but we could warn them (or warn the dev channel in general)

**Review by lithrel — APPROVED** [review]
:+1:
